src/azure-cli/azure/cli/command_modules/backup/tests/latest/test_workload_commands.py

```diff
@@ -561,13 +561,144 @@ def test_backup_wl_sql_restore_as_files(self):
+            self.check("resourceGroup", '{rg}'),
+            self.check("properties.recoveryPointMoveReadinessInfo.ArchivedRP.isReadyForMove", '{rp1_is_ready_for_move}')
+        ])
+
+        # Get Archived Recovery Points 
+        self.cmd('backup recoverypoint list -g {rg} -v {vault} -i {item} -c {container} --backup-management-type AzureWorkload --tier {rp1_tier} --query [0]', checks=[
+            self.check("tierType", '{rp1_tier}'),
+            self.check("resourceGroup", '{rg}')
+        ])
+
+        is_move = False
+        for i in rp_names:
+            if i['tierType']=="VaultStandard" and i['properties']['recoveryPointMoveReadinessInfo']['ArchivedRP']['isReadyForMove']==True:
+                self.kwargs['rp_move'] = i['name']
+                is_move = True
+                break
+        
+        if is_move:
+            # # Move Recovery points
+            self.cmd('backup recoverypoint move -g {rg} -v {vault} -i {item} -c {container} --source-tier VaultStandard --destination-tier VaultArchive --name {rp_move}', checks=[
+                self.check("properties.entityFriendlyName", 'msdb [archsqlccyvm2]'),
+                self.check("resourceGroup", '{rg}'),
+                self.check("properties.operation", "MoveRecoveryPoint"),
+                self.check("properties.status", "Completed")
+            ])
+        
+        is_restorable = False
+        for i in rp_names:
+            if i['tierType']=="VaultArchive":
+                self.kwargs['rp_restore'] = i['name']
+                is_restorable = True
+                break
+        
+        if is_restorable:
+            # # Integrated Restore
+            self.kwargs['rc'] = json.dumps(self.cmd('backup recoveryconfig show --vault-name {vault} -g {rg} --restore-mode OriginalWorkloadRestore --item-name {item} --container-name {container} --rp-name {rp_restore}').get_output_in_json(), separators=(',', ':'))
+            with open("recoveryconfig.json", "w") as f:
+                f.write(self.kwargs['rc'])
+
+            # # Trigger Restore
+            self.cmd('backup restore restore-azurewl -g {rg} -v {vault} --recovery-config recoveryconfig.json --rehydration-priority High', checks=[
+                self.check("properties.operation", "RestoreWithRehydrate"),
+                self.check("properties.status", "InProgress"),
+                self.check("resourceGroup", '{rg}')
+            ]).get_output_in_json()
+
     # SAP HANA workload tests start here
     # Please make sure you have the following setup in place before running the tests -
 
     # For the tests using akneema-hana-ccy and akneema-vault-ccy -
     # Each test will register the container at the start and unregister at the end of the test
     # Make sure that the container is not already registered since the start of the test
 
+    @record_only()
+    def test_workload_archive_hana (self):
```

Comment:
Change the test name to test_backup_wl_hana_archive

Reply:
dONE